sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/BlobCheckpointStore/BlobsCheckpointStore.cs

```diff
@@ -614,14 +614,25 @@ private static void TryReadLegacyCheckpoint(Span<byte> data, out long? offset, o
                     switch (jsonReader.GetString())
                     {
                         case "Offset":
-                            if (!jsonReader.Read() ||
-                                jsonReader.GetString() is not string offsetString ||
-                                !long.TryParse(offsetString, out long offsetValue))
+
+                            if (!jsonReader.Read())
                             {
                                 return;
                             }
 
-                            offset = offsetValue;
+                            var offsetString = jsonReader.GetString();
+                            if (offsetString != null)
```

Comment:
So the change is that if it is not a string we will continue rather than return?

Reply:
Only if it's a null value. If it's not a string or a null value we fail